Cubical/HITs/Sn/Properties.agda

```diff
@@ -13,29 +12,329 @@ open import Cubical.Foundations.Prelude
+
+  S2+n = S₊ (2 + n)
+  4n+2 = (2 + n) + (2 + n)
+
+  module WC-S (p : north ≡ north) where
+    P : (a b : S₊ (2 + n)) → Type₀
+    P a b = σ b ≡ p → hLevelTrunc 4n+2 (fiber (λ x → merid x ∙ merid a ⁻¹) p)
+
+    hLevelP : (a b : S₊ (2 + n)) → isOfHLevel 4n+2 (P a b)
+    hLevelP _ _ = isOfHLevelΠ 4n+2 λ _ → isOfHLevelTrunc 4n+2
+
+    leftFun : (a : S₊ (2 + n)) → P a north
+    leftFun a r = ∣ a , (rCancel' (merid a) ∙ rCancel' (merid north) ⁻¹) ∙ r ∣
+
+    rightFun : (b : S₊ (2 + n)) → P north b
+    rightFun b r = ∣ b , r ∣
+
+    funsAgree : leftFun north ≡ rightFun north
+    funsAgree =
+      funExt λ r →
+        cong′ (λ w → ∣ north , w ∣)
+          (cong (_∙ r) (rCancel' (rCancel' (merid north)))
+           ∙ lUnit r ⁻¹)
+
+    totalFun : (a b : S2+n) → P a b
+    totalFun =  wedgeConSn (suc n) (suc n) hLevelP rightFun leftFun funsAgree .fst
+
+    leftId : (λ x → totalFun x north) ≡ leftFun
+    leftId x i = wedgeConSn (suc n) (suc n) hLevelP rightFun leftFun funsAgree .snd .snd i x
+
+  fwd : (p : north ≡ north) (a : S2+n)
+    → hLevelTrunc 4n+2 (fiber σ p)
+    → hLevelTrunc 4n+2 (fiber (λ x → merid x ∙ merid a ⁻¹) p)
+  fwd p a = rec (isOfHLevelTrunc 4n+2) (uncurry (WC-S.totalFun p a))
+
+  fwdnorth : (p : north ≡ north) → fwd p north ≡ idfun _
+  fwdnorth p = funExt (elim (λ _ → isOfHLevelPath 4n+2 (isOfHLevelTrunc 4n+2) _ _)
+                      λ p → refl)
+
+  isEquivFwd : (p : north ≡ north) (a : S2+n) → isEquiv (fwd p a)
+  isEquivFwd p =
+    suspToPropElim (ptSn (suc n))
+                   (λ _ → isPropIsEquiv _)
+                   helper
+    where
+    helper : isEquiv (fwd p north)
+    helper = subst isEquiv (sym (fwdnorth p)) (idIsEquiv _)
+
+  interpolate : (a : S2+n)
+          → PathP (λ i → S2+n → north ≡ merid a i) (λ x → merid x ∙ merid a ⁻¹) merid
+  interpolate a i x j = compPath-filler (merid x) (merid a ⁻¹) (~ i) j
+
+  Code : (y : Susp S2+n) → north ≡ y → Type₀
+  Code north p = hLevelTrunc 4n+2 (fiber σ p)
+  Code south q = hLevelTrunc 4n+2 (fiber merid q)
+  Code (merid a i) p =
+    Glue
+      (hLevelTrunc 4n+2 (fiber (interpolate a i) p))
+      (λ
+        { (i = i0) → _ , (fwd p a , isEquivFwd p a)
+        ; (i = i1) → _ , idEquiv _
+        })
+
+  encode' : (y : S₊ (3 + n)) (p : north ≡ y) → Code y p
+  encode' y = J Code ∣ north , rCancel' (merid north) ∣
+
+  encodeMerid : (a : S2+n) → encode' south (merid a) ≡ ∣ a , refl ∣
+  encodeMerid a =
+    cong (transport (λ i → gluePath i))
+      (funExt⁻ (funExt⁻ (WC-S.leftId refl) a) _ ∙ λ i → ∣ a , lem (rCancel' (merid a)) (rCancel' (merid north)) i ∣)
+    ∙ transport (PathP≡Path gluePath _ _)
+      (λ i → ∣ a , (λ j k → rCancel-filler' (merid a) i j k) ∣)
+    where
+    gluePath : I → Type _
+    gluePath i = hLevelTrunc 4n+2 (fiber (interpolate a i) (λ j → merid a (i ∧ j)))
+
+    lem : ∀ {ℓ} {A : Type ℓ} {x y z : A} (p : x ≡ y) (q : z ≡ y) → (p ∙ q ⁻¹) ∙ q ≡ p
+    lem p q = assoc p (q ⁻¹) q ⁻¹ ∙ cong (p ∙_) (lCancel q) ∙ rUnit p ⁻¹
```

Comment:
Can't you use `_∙∙_∙∙_`  ?

Reply:
Oh right. I'll change that in Evan's original proof too then

Reply:
Great, please fix both!